govcd/nsxt_alb_service_engine_group_assignment_test.go

```diff
@@ -0,0 +1,155 @@
+	albSettingsConfig := &types.NsxtAlbConfig{
+		Enabled: true,
+	}
+	enabledSettings, err := edge.UpdateAlbSettings(albSettingsConfig)
+	check.Assert(err, IsNil)
+	check.Assert(enabledSettings.Enabled, Equals, true)
+	PrependToCleanupList("OpenApiEntityAlbSettingsDisable", "OpenApiEntityAlbSettingsDisable", edge.EdgeGateway.Name, check.TestName())
+
+	serviceEngineGroupAssignmentConfig := &types.NsxtAlbServiceEngineGroupAssignment{
+		GatewayRef:            &types.OpenApiReference{ID: edge.EdgeGateway.ID},
+		ServiceEngineGroupRef: &types.OpenApiReference{ID: seGroup.NsxtAlbServiceEngineGroup.ID},
+	}
+	assignment, err := vcd.client.CreateAlbServiceEngineGroupAssignment(serviceEngineGroupAssignmentConfig)
+	check.Assert(err, IsNil)
+	check.Assert(assignment.NsxtAlbServiceEngineGroupAssignment.ID, Not(Equals), "")
+	openApiEndpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointAlbServiceEngineGroupAssignments + assignment.NsxtAlbServiceEngineGroupAssignment.ID
+	PrependToCleanupListOpenApi(assignment.NsxtAlbServiceEngineGroupAssignment.ServiceEngineGroupRef.Name, check.TestName(), openApiEndpoint)
+
+	// Get By ID
+	assignmentById, err := vcd.client.GetAlbServiceEngineGroupAssignmentById(assignment.NsxtAlbServiceEngineGroupAssignment.ID)
+	check.Assert(err, IsNil)
+	check.Assert(assignmentById.NsxtAlbServiceEngineGroupAssignment, DeepEquals, assignment.NsxtAlbServiceEngineGroupAssignment)
+
+	// Get By Name
+	assignmentByName, err := vcd.client.GetAlbServiceEngineGroupAssignmentByName(assignment.NsxtAlbServiceEngineGroupAssignment.ServiceEngineGroupRef.Name)
+	check.Assert(err, IsNil)
+	check.Assert(assignmentByName.NsxtAlbServiceEngineGroupAssignment, DeepEquals, assignment.NsxtAlbServiceEngineGroupAssignment)
+
+	// Get all
+	allAssignments, err := vcd.client.GetAllAlbServiceEngineGroupAssignments(nil)
+	check.Assert(err, IsNil)
+	var foundAssignment bool
+	for i := range allAssignments {
+		if allAssignments[i].NsxtAlbServiceEngineGroupAssignment.ID == assignment.NsxtAlbServiceEngineGroupAssignment.ID {
+			foundAssignment = true
+		}
+	}
+	check.Assert(foundAssignment, Equals, true)
+
+	assignment.NsxtAlbServiceEngineGroupAssignment.MaxVirtualServices = takeIntAddress(50)
+	assignment.NsxtAlbServiceEngineGroupAssignment.MinVirtualServices = takeIntAddress(30)
+	// Expect an error because "DEDICATED" service engine group does not support specifying virtual services
+	updatedAssignment, err := assignment.Update(assignment.NsxtAlbServiceEngineGroupAssignment)
+	check.Assert(err, NotNil)
+	check.Assert(updatedAssignment, IsNil)
+
+	// Perform immediate cleanups
+	err = assignment.Delete()
+	check.Assert(err, IsNil)
+	err = edge.DisableAlb()
+	check.Assert(err, IsNil)
+	err = seGroup.Delete()
+	check.Assert(err, IsNil)
+	err = cloud.Delete()
+	check.Assert(err, IsNil)
+	err = controller.Delete()
+	check.Assert(err, IsNil)
+}
+
+func (vcd *TestVCD) Test_GetAllEdgeAlbServiceEngineGroupAssignmentsSharedSe(check *C) {
```

Comment:
Test_GetAllEdgeAlbServiceEngineGroupAssignmentsShared**Se**(check *C) {
 what this Se means? maybe we could add comment or change name?

Reply:
> Test_GetAllEdgeAlbServiceEngineGroupAssignmentsShared**Se**(check *C) { what this Se means? maybe we could add comment or change name?


Se is "Service Engine". In fact we can remove it at all and leave the "Shared" word for meaning.



`Test_GetAllEdgeAlbServiceEngineGroupAssignmentsDedicated` and `Test_GetAllEdgeAlbServiceEngineGroupAssignmentsShared`